From f0ec0358d113e323230b741bfb3062240ffe9903 Mon Sep 17 00:00:00 2001
From: kumarama-i <aman.kumar@intel.com>
Date: Mon, 21 Feb 2022 15:16:36 +0530
Subject: [PATCH 1/1] m_can_pci: Update RTPM related APIs

This fix is to get the proper base address of CAN bus which enable
suspend and resume feature of it.

Signed-off-by: Aman Kumar <aman.kumar@intel.com>
---
 drivers/net/can/m_can/m_can_pci.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/net/can/m_can/m_can_pci.c b/drivers/net/can/m_can/m_can_pci.c
index aa603d58e8dd..b56f62031bac 100644
--- a/drivers/net/can/m_can/m_can_pci.c
+++ b/drivers/net/can/m_can/m_can_pci.c
@@ -224,9 +224,7 @@ static __maybe_unused int m_can_pci_resume(struct device *dev)
 
 static int __maybe_unused m_can_pci_runtime_suspend(struct device *dev)
 {
-	struct pci_dev *pdev = container_of(dev, struct pci_dev, dev);
-	struct net_device *ndev = pci_get_drvdata(pdev);
-	struct m_can_classdev *mcan_class = netdev_priv(ndev);
+	struct m_can_classdev *mcan_class = dev_get_drvdata(dev);
 	u32 d0i3c_reg;
 	u32 cgsr_reg;
 	unsigned long j0, j1, delay;
@@ -261,9 +259,7 @@ static int __maybe_unused m_can_pci_runtime_suspend(struct device *dev)
 
 static int __maybe_unused m_can_pci_runtime_resume(struct device *dev)
 {
-	struct pci_dev *pdev = container_of(dev, struct pci_dev, dev);
-	struct net_device *ndev = pci_get_drvdata(pdev);
-	struct m_can_classdev *mcan_class = netdev_priv(ndev);
+	struct m_can_classdev *mcan_class = dev_get_drvdata(dev);
 	u32 d0i3c_reg;
 	u32 cgsr_reg;
 
-- 
2.17.1

